hack/create-cert.sh

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/env bash
+
+# Copyright 2020 PingCAP, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+usage() {
+    cat <<EOF
+create-cert.sh
+create certification for pump and drainer conponent
+
+  --namespace    the namespace of component belongs
+  --component    name of component (pump, drainer)
+  --cluster      cluster name of component belongs
+EOF
+    exit 1
+}
+
+while [[ $# -gt 0 ]]; do
+    case ${1} in
+        --namespace)
+            namespace="$2"
+            shift
+            ;;
+        --component)
+            component="$2"
+            shift
+            ;;
+        --cluster)
+            cluster="$2"
+            shift
+            ;;
+        *)
+            usage
+            ;;
+    esac
+    shift
+done
+
+if [ -z ${namespace} ]; then
+	usage
+fi
+
+if [ -z ${component} ]; then
+	usage
+fi
+
+if [ -z ${cluster} ]; then
+	usage
+fi
+
+service=${cluster}-${component}
+secret=${cluster}-${namespace}-${component}
+
+if [ ! -x "$(command -v openssl)" ]; then
+    echo "openssl not found"
+    exit 1
+fi
+
+csrName=${service}.${namespace}
+tmpdir=$(mktemp -d)
+
+echo ${tmpdir}
+
+cat <<EOF >> ${tmpdir}/csr.conf
+[req]
+req_extensions = v3_req
+distinguished_name = req_distinguished_name
+[req_distinguished_name]
+[ v3_req ]
+basicConstraints = CA:FALSE
+keyUsage = nonRepudiation, digitalSignature, keyEncipherment
+extendedKeyUsage = serverAuth
+subjectAltName = @alt_names
+[alt_names]
+DNS.1 = ${service}
+DNS.2 = ${service}.${namespace}
+DNS.3 = ${service}.${namespace}.svc
+DNS.4 = *.${service}
+DNS.5 = *.${service}.${namespace}
+DNS.5 = *.${service}.${namespace}.svc
+IP.1 = 127.0.0.1
+EOF
+
+openssl genrsa -out ${tmpdir}/server-key.pem 2048
+openssl req -new -key ${tmpdir}/server-key.pem -subj "/CN=${service}.${namespace}.svc" -out ${tmpdir}/server.csr -config ${tmpdir}/csr.conf
+
+ # clean-up any previously created CSR for our service. Ignore errors if not present.
+kubectl delete csr ${csrName} 2>/dev/null || true
+
+ # create  server cert/key CSR and  send to k8s API
+cat <<EOF | kubectl create -f -
+apiVersion: certificates.k8s.io/v1beta1
+kind: CertificateSigningRequest
+metadata:
+  name: ${csrName}
+spec:
+  groups:
+  - system:authenticated
+  request: $(cat ${tmpdir}/server.csr | base64 | tr -d '\n')
+  usages:
+  - digital signature
+  - key encipherment
+  - server auth
+  - client auth
+EOF
+
+ # verify CSR has been created
+while true; do
+    kubectl get csr ${csrName}
```

Comment:
Add log here and sleep sometime before retry.

Reply:
I dont't think this is neccessary

Reply:
With log here, we can know what this script is doing now.
And we can sleep 1s before the next check to reduce the resource usage and API calls.